Tasks/DeployVisualStudioTestAgent/DownloadTestAgent.ps1

```diff
@@ -0,0 +1,49 @@
+# DownloadTestAgent.ps1 takes two parameters , sourcePath and destinationPath
+# $sourcePath is the path from which the test agent is to be downloaded or copied.
+# $destinationPath is the location to which the test agent will be downloaded or copied.
+
+# Validate that the given source path exists and is not a directory.
+function ValidateSourceFile([string] $sourcePath)
+{
+   if(! (Test-Path -Path $sourcePath))
+   {
+         throw "Test agent source path '{0}' is not accessible to the test machine. Please check if the file exists and that test machine has access to that machine" -f $sourcePath
```

Comment:
Please look at conversation section. I had given some comments on this line reviewing individual commits as part of PR.